src/core/shred.zig

```diff
@@ -11,4 +13,24 @@ pub const ShredVersion = struct {
     pub fn init_manually_set(version: u16) Self {
         return Self{ .value = version };
     }
+
+    pub fn version_from_hash(hash: *Hash) u16 {
+        const hash_bytes = hash.data;
+        var accum: [2]u8 = .{ 0, 0 };
+        var chunks = std.mem.window(u8, &hash_bytes, 2, 2);
+
+        while (chunks.next()) |chunk| {
+            for (chunk, accum[0..chunk.len]) |ch, *acc| {
```

Comment:
can we unwrap this for loop so we just do `acc[0] ^= chunk[0]` for indexs 0 and 1 to be more clear